mailet/api/src/main/java/org/apache/mailet/LoopPrevention.java

```diff
@@ -0,0 +1,66 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.mailet;
+
+import java.util.Collection;
+import java.util.Set;
+
+import org.apache.james.core.MailAddress;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+public class LoopPrevention {
```

Comment:
Please unit test this class

Reply:
Is it necessary to write unit test for a method like this?

```
public static Set<MailAddress> nonRecordedRecipients(Set<MailAddress> recipients, Set<MailAddress> recordedRecipients) {
        return Sets.difference(recipients, recordedRecipients);
    }
```

Reply:
Please write unit test for the relevant methods.

> Is it necessary to write unit test for a method like this?

Is it necessary to write a method like this in the first place?

Reply:
added